packages/extension-commerce-bm-seo/src/components/with-optional-commerce-sdk-react-provider.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {getAppOrigin} from '@salesforce/pwa-kit-react-sdk/utils/url'
+import {CommerceApiProvider, useCommerceApi} from '@salesforce/commerce-sdk-react'
+import {UserConfig} from '../types/config'
+import {logger} from '../logger'
+
+/**
+ * Checks if the CommerceApiProvider is already installed in the component tree.
+ * @returns boolean, true if the CommerceApiProvider is installed, false otherwise.
+ */
+const useHasCommerceApiProvider = () => {
+    let hasProvider = false
+
+    try {
+        const api = useCommerceApi()
+
+        // the api object is an object with a bunch of api clients like ShopperProduct, ShopperOrder, etc.
+        // if the object is empty, then the CommerceApiProvider is not installed
+        if (Object.keys(api).length > 0) {
+            hasProvider = true
+        }
+    } catch (_) {
+        hasProvider = false
+    }
+
+    return hasProvider
+}
```

Comment:
Move this to a hooks specific file please, it's possible to also follow the pattern that @hajinsuha1 had moving this to the extension sdk under a commerce folder.